### PR TITLE
server:  Warn in logs on admin mutations via GET. Mutations should be POSTed.

### PR DESCRIPTION
**Pull request by jmarantz**
Description:
When a mutating admin-console endpoint is reached, verify that it's a POST. For now, the operation succeeds but a warning log is printed.

This is a second attempt of #2912, which (a) contained an earlier version of the log-testing infra 
from that was submitted #2930, and (b) was impacted by a merged commit that lacked a DCO.

This is a step toward fixing #2763

Risk Level: Low

Testing: //test/...

Docs Changes:

Link to Data Plane PR]
[in progress, will update]

Release Notes: [in progress, will update]

If this change is user impacting you must add a release note via a discrete PR to
version_history.rst.

API Changes: https://github.com/envoyproxy/data-plane-api/pull/602

Deprecated:
Deprecates supporting admin mutations with a GET.

### CODE REVIEW

File: source/server/http/admin.cc [hunk 1]
```diff
@@ -758,7 +757,7 @@ void AdminImpl::createFilterChain(Http::FilterChainFactoryCallbacks& callbacks)
   callbacks.addStreamDecoderFilter(Http::StreamDecoderFilterSharedPtr{new AdminFilter(*this)});
 }
 
-Http::Code AdminImpl::runCallback(absl::string_view path_and_query,
+Http::Code AdminImpl::runCallback(absl::string_view path_and_query, AdminFilter* admin_filter,
```
Comment (mattklein123) [inline source/server/http/admin.cc]: nit: `AdminFilter& admin_filter` (can't be null)

Comment (jmarantz) [reply source/server/http/admin.cc]: done

File: source/server/http/admin.cc [hunk 2]
```diff
@@ -770,6 +769,14 @@ Http::Code AdminImpl::runCallback(absl::string_view path_and_query,
 
   for (const UrlHandler& handler : handlers_) {
     if (path_and_query.compare(0, query_index, handler.prefix_) == 0) {
+      if (handler.mutates_server_state_) {
+        const absl::string_view method =
+            admin_filter.requestHeaders().Method()->value().getStringView();
```
Comment (mattklein123) [inline source/server/http/admin.cc]: Sorry I just noticed this. Is there any reason to pass the entire filter here vs. just passing the request headers? Seems a bit cleaner unless it's necessary (we can avoid accessor, etc.).

Comment (jmarantz) [reply source/server/http/admin.cc]: Good question. What was on my mind here was eliminating the need for the dynamic_cast in https://github.com/envoyproxy/envoy/pull/2736/files/b9a4301d5b885ec2c244315791768edd68fa8352 in #2736 .

It makes sense to me to have admin handlers get more request context in general.  And in the case of that PR,  what's needed is ```Http::StreamDecoderFilterCallbacks* callbacks_```.  I'm also fine reducing the exposure now to just request-headers, and that could be broadened to the entire AdminFilter& in the future if needed.  I admit, though, I don't really know whether those callbacks_ are truly needed to solve the hystrix issues.

It's a question of speculative generality vs. reducing the need for an antipated refactor in the future, and I'm fine either way.

Comment (mattklein123) [reply source/server/http/admin.cc]: My preference is to keep it simple for now unless there is a proven need, but I don't feel that strongly about it.

Comment (jmarantz) [reply source/server/http/admin.cc]: ok done.

File: source/server/http/admin.h [hunk 3]
```diff
@@ -31,6 +31,8 @@
 namespace Envoy {
 namespace Server {
 
+class AdminFilter;
```
Comment (mattklein123) [inline source/server/http/admin.h]: nit: not needed

**Pull request merged**